site/src/pages/DeploymentSettingsPage/OAuth2AppsSettingsPage/OAuth2AppForm.tsx

```diff
@@ -54,17 +88,54 @@ export const OAuth2AppForm: FC<OAuth2AppFormProps> = ({
 					autoFocus
 					fullWidth
 				/>
-				<TextField
-					name="callback_url"
-					label="Callback URL"
-					defaultValue={app?.callback_url}
-					error={Boolean(apiValidationErrors?.callback_url)}
-					helperText={
-						apiValidationErrors?.callback_url ||
-						"The full URL to redirect to after a user authorizes an installation."
-					}
-					fullWidth
-				/>
+
+				{/* Redirect URIs Section */}
+				<Stack spacing={1}>
+					<div css={{ display: "flex", alignItems: "center", gap: 8 }}>
+						<span css={{ fontWeight: 500, fontSize: 14 }}>Redirect URIs</span>
+						<IconButton
+							size="small"
+							onClick={addRedirectUri}
+							css={{ padding: 4 }}
+							title="Add redirect URI"
+						>
+							<AddIcon fontSize="small" />
+						</IconButton>
+					</div>
+
+					{redirectUris.map((uri, index) => (
+						<div
+							key={index}
+							css={{ display: "flex", gap: 8, alignItems: "flex-start" }}
+						>
+							<TextField
+								value={uri}
+								onChange={(event) =>
+									updateRedirectUri(index, event.target.value)
+								}
+								placeholder="https://example.com/callback"
+								error={Boolean(apiValidationErrors?.redirect_uris)}
+								helperText={
+									index === 0 &&
+									(apiValidationErrors?.redirect_uris ||
+										"Full URLs where users will be redirected after authorization. At least one is required.")
+								}
+								fullWidth
+							/>
+							{redirectUris.length > 1 && (
+								<IconButton
+									size="small"
+									onClick={() => removeRedirectUri(index)}
+									css={{ padding: 4, marginTop: index === 0 ? 8 : 0 }}
+									title="Remove redirect URI"
+								>
+									<DeleteIcon fontSize="small" />
+								</IconButton>
+							)}
```

Comment:
you could use a `Stack` instead of needing to conditionally apply `margin-top`, which uses the more modern `gap`